core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -434,13 +472,14 @@ class GroupMetadataManager(brokerId: Int,
 
         // loop breaks if leader changes at any time during the load, since getHighWatermark is -1
         val loadedOffsets = mutable.Map[GroupTopicPartition, OffsetAndMetadata]()
-        val removedOffsets = mutable.Set[GroupTopicPartition]()
+        val pendingOffsets = mutable.Map[Long, mutable.Map[GroupTopicPartition, OffsetAndMetadata]]()
         val loadedGroups = mutable.Map[String, GroupMetadata]()
         val removedGroups = mutable.Set[String]()
+        val producerGroups = mutable.Map[Long, mutable.Set[String]]()
 
         while (currOffset < highWaterMark && !shuttingDown.get()) {
           val fetchDataInfo = log.read(currOffset, config.loadBufferSize, maxOffset = None, minOneMessage = true,
-            isolationLevel = IsolationLevel.READ_UNCOMMITTED)
+            isolationLevel = IsolationLevel.READ_COMMITTED)
```

Comment:
Hmm... why did we change this? I think we're expecting to read even the incomplete transactions so that we can populate the pending transaction offset commits in `GroupMetadata`.